### PR TITLE
Fix the version string and release 0.4.1

### PR DESCRIPTION
**Pull request by 7fELF**
Vault introduced a versioning system for plugins. The plugin must reports its version. vault-plugin-secrets-exoscale implemented this versioning in 0.4.0.

It seems like vault requires the version string to start with the "v" prefix, we didn't do that in "0.4.0".

This PR adds the v prefix and releases a new version of the plugin.

If you installed 0.4.0 you might not be able to unregister it: If you installed the plugin using `vault plugin register` without specifying the `-version` flag, vault automatically retrieves the version number and uses it without validating the "v" prefix. The API call to unregister plugin is however not capable of removing a versioned plugin that doesn't have the "v" prefix. If this is your case, you can use [sys/raw](https://developer.hashicorp.com/vault/api-docs/system/raw) to remove the bogus version of the plugin:

```
# ⚠️ WARNING ⚠️ 
# Please backup vault and read the `sys/raw` documentation before doing this
# Editing `sys/raw` manually can break your vault cluster permanently

vault delete -format=json /sys/raw/core/plugin-catalog/secret/exoscale/0.4.0
```

If you didn't install 0.4.0, you can upgrade directly to 0.4.1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,107 +2,8 @@
 
 [![Actions Status](https://github.com/exoscale/vault-plugin-secrets-exoscale/workflows/CI/badge.svg)](https://github.com/exoscale/vault-plugin-secrets-exoscale/actions?query=workflow%3ACI)
 
-This is a [backend plugin][vaultdocplugins] plugin to be used with HashiCorp [Vault](https://www.vaultproject.io/). This plugin generates Exoscale IAM API keys which can be restricted to specific operations according to predefined roles.
+This is a [HashiCorp Vault](https://www.vaultproject.io/) secrets engine that generates [Exoscale IAM API keys](https://community.exoscale.com/documentation/iam/) with specific [IAM Roles](https://community.exoscale.com/documentation/iam/iam-api-key-roles-policies/#iam-roles).
 
-This guide assumes you have already installed Vault and have a basic understanding of how Vault works. Otherwise, first read this guide on how to [get started with Vault][vaultdocintro].
+You can find the official documentation of this plugin on [the wiki](https://github.com/exoscale/vault-plugin-secrets-exoscale/wiki) of this repo.
```
Comment (7fELF) [inline README.md:7]: The doc is similar to the README for now, but the idea is to make it evolve into something more detailed and better organized (getting started, build from source, install, upgrade, manage IAMv3 roles etc.)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,14 @@
 # Changelog
 
+## 0.4.1
+
+### Bug Fixes
+- Fix the version string to add the "v" prefix e.g v0.4.1 instead of 0.4.0 [#20](https://github.com/exoscale/vault-plugin-secrets-exoscale/pull/20)
 
 ## 0.4.0
 
+⚠️  This version has [a bug related to plugin versionning](https://github.com/exoscale/vault-plugin-secrets-exoscale/pull/20)⚠️, it is recommended to skip it and install 0.4.1 directly ⚠️
```
Comment (7fELF) [inline CHANGELOG.md:10]: I removed the binaries for the [0.4.0 release](https://github.com/exoscale/vault-plugin-secrets-exoscale/releases/tag/v0.4.0) and added a warning.

**Review by irregulator — APPROVED** [review]
Thanks Antoine.

**Review by yannprvst — APPROVED** [review]
\o/

**Pull request merged**